### ISSUE TITLE
Ignoring my settings for CP or IV

### ISSUE THREAD

**Issue by catsandcodes** [NECROBOTIO/NecroBot#2364]
So far I have like 10 snorlax's either they meet the requirement for both but i do have set the duplicate pokemon to just 1, shouldnt it detect if IV is a priority and duplicate to 1, transfer all snorlax and leave the one with the strongest IV? it keeps hoarding up on them. any tips.

**Comment by actislav** [comment]
every pokemon meet the requirement will be kept

**Comment by catsandcodes** [comment]
actually, I have figured it out, it was one of the pokemons in the DONOTTRANSFER list.. if its there the program wont transfer it at all.

**Comment by Seyrenvox** [comment]
the bot ignoring my settings, tho.
i have 7 Cloysters. none of them met the requirements i've set in "PokemonsTransferFilter" (nor the global settings).
Cloyster not on my "NotToTransferList"

**Comment by xeniusw** [comment]
This is the reason: https://github.com/NECROBOTIO/NecroBot/pull/1896
As he says the logic is to keep "or" if you do "and" it will keep CP and IV, I don't know how to set the moves up properly yet as it gives me an error, if anybody figures it out it'll be much appreciated.

I've tried stuff like this without luck

    "Machamp": {
      "KeepMinCp": 1700,
      "KeepMinIvPercentage": 90.0,
      "KeepMinDuplicatePokemon": 2,
      "Moves": [KarateChopFast],[CrossChop],
      "KeepMinOperator": "or"

**Comment by Seyrenvox** [comment]
okay, that's explain why.

this is my setting for Golduck
```
"Golduck": {
      "KeepMinCp": 2300,
      "KeepMinIvPercentage": 93.0,
      "KeepMinDuplicatePokemon": 1,
      "Moves": [
        "waterGunFast",
        "hydroPump"
      ],
      "KeepMinOperator": "and"
    },
```
as you can see, i'm using an AND operator in "KeepMinOperator". so, it should keep Golduck with minimal IV 93 **AND** 2300 CP.
but this is my Golducks:
```
Golduck	30	1830 	28.93403649	122	122	ConfusionFast	HydroPump	18
Golduck	30	1818 	25.08747238	119	119	WaterGunFast	HydroPump	18
Golduck	28	1803 	62.62178244	115	115	ConfusionFast	HydroPump	18
Golduck	29	1801 	39.81361114	125	125	ConfusionFast	HydroPump	18
Golduck	29	1793 	37.22703945	120	120	WaterGunFast	Psychic	18
Golduck	29	1793 	37.22703945	120	120	WaterGunFast	HydroPump	18
Golduck	30	1789 	15.53075806	120	120	WaterGunFast	IceBeam	18
```
none of those met the minimal requirement for CP (third column), or IV (fourth column). BUT, they met (at least one of) the "Moves" requirement.
in other words, the "Moves" setting will override the "KeepMinOperator", and that's not what i actually expected

**Comment by catsandcodes** [comment]
oh wow, i actually havent played around with that much filter, for now im just using the beginning keeps and transfer. 

closing this issue guys

**Comment by simplyphp** [comment]
Resolved in #2367
